src/relentless/optimize/objective.py

```diff
@@ -374,6 +389,284 @@ def compute(self, variables, directory=None):
+                for i in snap.particles.types:
+                    type_masks[i] = snap.particles.typeid == snap.particles.types.index(
+                        i
+                    )
+
+                bonded_exclusions = self.potentials.pair.exclusions
+                if bonded_exclusions is not None:
+                    if (
+                        "1-2" in bonded_exclusions
+                        and snap.bonds.N != 0
+                        and len(neighbors[:]) > 0
+                    ):
+                        bonds = numpy.vstack(
+                            [snap.bonds.group, numpy.flip(snap.bonds.group, axis=1)],
+                        )
+
+                        bond_exclusion_filter = EnsembleAverage._cantor_pairing(
+                            self, bonds, neighbors
+                        )
+
+                        neighbors.filter(bond_exclusion_filter)
+                    if (
+                        "1-3" in bonded_exclusions
+                        and snap.angles.N != 0
+                        and len(neighbors[:]) > 0
+                    ):
+                        angles = numpy.vstack(
+                            [snap.angles.group, numpy.flip(snap.angles.group, axis=1)],
+                        )
+
+                        angle_exclusion_filter = EnsembleAverage._cantor_pairing(
+                            self, angles[:, (0, -1)], neighbors
+                        )
+
+                        neighbors.filter(angle_exclusion_filter)
+                    if (
+                        "1-4" in bonded_exclusions
+                        and snap.dihedrals.N != 0
+                        and len(neighbors[:]) > 0
+                    ):
+                        dihedrals = numpy.vstack(
+                            [
+                                snap.dihedrals.group,
+                                numpy.flip(snap.dihedrals.group, axis=1),
+                            ],
+                        )
+
+                        dihedral_exclusion_filter = EnsembleAverage._cantor_pairing(
+                            self, dihedrals[:, (0, -1)], neighbors
+                        )
+
+                        neighbors.filter(dihedral_exclusion_filter)
+
+                filter_j_gt_i = neighbors[:, 1] > neighbors[:, 0]
+                neighbors.filter(filter_j_gt_i)
+                # pair contributions to the gradient
+                for i in snap.particles.types:
+                    for j in snap.particles.types:
+                        filter_ij = numpy.logical_and(
+                            type_masks[i][neighbors[:, 0]],
+                            type_masks[j][neighbors[:, 1]],
+                        )
+                        for var in variables:
+                            gradient[var] += numpy.sum(
+                                self.potentials.pair.derivative(
+                                    (i, j), var, x=neighbors.distances[filter_ij]
```

Comment:
The distance freud reports respect the PBCs, right?

Reply:
Yes